client/devices/client.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package devices
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/mendersoftware/mender-cli/log"
+)
+
+type devicesList struct {
+	devices []deviceData
+}
+
+type deviceData struct {
+	ID           string `json:"id"`
+	IdentityData struct {
+		Mac string `json:"mac"`
+		Sku string `json:"sku"`
+		Sn  string `json:"sn"`
+	} `json:"identity_data"`
+	Status    string `json:"status"`
+	CreatedTs string `json:"created_ts"`
+	UpdatedTs string `json:"updated_ts"`
+	AuthSets  []struct {
+		ID           string `json:"id"`
+		PubKey       string `json:"pubkey"`
+		IdentityData struct {
+			Mac string `json:"mac"`
+			Sku string `json:"sku"`
+			Sn  string `json:"sn"`
+		} `json:"identity_data"`
+		Status string `json:"status"`
+		Ts     string `json:"ts"`
+	} `json:"auth_sets"`
+	Decommissioning bool `json:"decommissioning"`
+}
+
+const (
+	devicesListURL = "/api/management/v2/devauth/devices"
+)
+
+type Client struct {
+	url            string
+	devicesListURL string
+	client         *http.Client
+}
+
+func NewClient(url string, skipVerify bool) *Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
+
+	return &Client{
+		url:            url,
+		devicesListURL: JoinURL(url, devicesListURL),
+		client: &http.Client{
+			Transport: tr,
+		},
+	}
+}
+
+func (c *Client) ListDevices(tokenPath string, detailLevel int) error {
```

Comment:
Musing:
with the amount of code shared with `/deployments/client` it should be possible to abstract away most of the HTTP request creation. OTOH, I don't really see a huge benefit to it, what do you think?

Reply:
I'll try to refactor a bit :)